cmd/bnsd/app/datamigration.go

```diff
@@ -16,57 +17,82 @@ import (
 )
 
 func init() {
-	technicalExecutors, err := weave.ParseAddress("cond:gov/rule/0000000000000003")
-	if err != nil {
-		panic(err)
-	}
-	governingBoard, err := weave.ParseAddress("cond:gov/rule/0000000000000001")
-	if err != nil {
-		panic(err)
-	}
-
 	datamigration.MustRegister("initialize x/msgfee configuration owner", datamigration.Migration{
 		RequiredSigners: []weave.Address{governingBoard},
 		ChainIDs: []string{
 			"iov-mainnet",
 		},
-		Migrate: func(ctx context.Context, db weave.KVStore) error {
-			var conf msgfee.Configuration
-			switch err := gconf.Load(db, "msgfee", &msgfee.Configuration{}); {
-			case errors.ErrNotFound.Is(err):
-				conf.Metadata = &weave.Metadata{Schema: 1}
-				conf.Owner = technicalExecutors
-			case err == nil:
-				if len(conf.Owner) != 0 {
-					return errors.Wrap(errors.ErrState, "configuration owner already set")
-				}
-				conf.Owner = technicalExecutors
-			default:
-				return errors.Wrap(err, "load")
-			}
-
-			if err := gconf.Save(db, "msgfee", &conf); err != nil {
-				return errors.Wrap(err, "save")
-			}
-			return nil
-		},
+		Migrate: initializeMsgfeeConfiguration,
 	})
-
 	datamigration.MustRegister("rewrite username accounts", datamigration.Migration{
 		RequiredSigners: []weave.Address{governingBoard},
 		ChainIDs: []string{
 			"iov-mainnet",
 		},
 		Migrate: rewriteUsernameAccounts,
 	})
+	datamigration.MustRegister("initialize preregistration configuration", datamigration.Migration{
+		RequiredSigners: []weave.Address{governingBoard},
+		ChainIDs: []string{
+			"iov-mainnet",
+		},
+		Migrate: initializePreregistrationConfiguration,
+	})
+	datamigration.MustRegister("rewrite preregistration records", datamigration.Migration{
+		RequiredSigners: []weave.Address{governingBoard},
+		ChainIDs: []string{
+			"iov-mainnet",
+		},
+		Migrate: rewritePreregistrationRecords,
+	})
 }
 
-func rewriteUsernameAccounts(ctx context.Context, db weave.KVStore) error {
-	governingBoard, err := weave.ParseAddress("cond:gov/rule/0000000000000001")
+var (
+	governingBoard     = mustParse("cond:gov/rule/0000000000000001")
+	technicalExecutors = mustParse("cond:gov/rule/0000000000000003")
+)
+
+func mustParse(encodedAddress string) weave.Address {
+	a, err := weave.ParseAddress(encodedAddress)
 	if err != nil {
 		panic(err)
 	}
+	return a
+}
 
+func initializePreregistrationConfiguration(ctx context.Context, db weave.KVStore) error {
+	conf := preregistration.Configuration{
+		Metadata: &weave.Metadata{Schema: 1},
+		Owner:    technicalExecutors,
```

Comment:
@davepuchyr preregistration configuration is initialized with `technicalExecututors` as the admin. Configuration admin signature is required for registering a new record. Is this setup acceptable?

Reply:
No, we want to use a throw away address for the addition of records.

@webmaster128 , will it make IOV Core's and the front-end team's life easier if the `governingBoard` initializes the configuration?

Reply:
`preregistration.Configuration` has no admin, just an owner, right?

I think we should decouple config ownership and the `writer` address which has write access to the records. Then `writer` can be a throw away address and the config can be owned by `technicalExecututors`.

Reply:
Correct. Right now I am using the same address for configuration administration and for authorizing the only message this extension provides. My reasoning for this is that you do not need to manage configuration, as there is nothing to configure. Configuration owner address will be populated once via migration and the extension should be used for few days only.

How should I change this setup?

Reply:
- Configuration is owned by `technicalExecututors` (as all the other configs as well)
- There is an independent throw away address that manages the list, which should be in the configuration. This is what I called `writer` above, but it can have any name that makes sense.
- Configuration is expeted to never change. So no Config update message required for now. But if necessary, a governance proposal could change the config (when Dave's key goes on 🔥).

Reply:
`preregistration` extension is a mean to import data. It is to be used for a day and forgotten after that. Why do you want to invest so much effort into all this complication, when a straightforward implementation fits well?

Reply:
The only effort I see here is this unnecessary discussion, which is a result of code that is unintuitive to read. To spare a few lines of dump static code, we re-use the config owner field in a way that has nothing to do with config ownership. You use the term "admin", which I don't see in code. Dave is asking for `governingBoard` because he does not see what config owner will be used for. This PR entirely misses a placeholder for the address that Dave will use to write all the preregistration entired. This clearly cannot be the `technicalExecututors`. 

So yes, I'd rather see code that is self-explanatiory than saving a few lines.

Reply:
> The only effort I see here is this unnecessary discussion

I think this summarizes it for now.